aiida_vasp/utils/vasp/ibrion.py

```diff
@@ -0,0 +1,107 @@
+
+    Usage::
+
+        from aiida_vasp.utils.vasp.ibrion import IbrionFlags, Ibrion
+
+        incar = {...}
+        ibrion = Ibrion(ion_updates=IbrionFlags.MOLECULAR_DYNAMICS)
+        incar.update(ibrion.param)
+        errors, warnings = ibrion.clean(incar)
+        for message in errors + warnings:
+            print message  # or log, etc
+        if errors:
+            raise ValueError(errors[0])
+    """
+
+    IBRION_EXPLANATIONS = {
+        IbrionFlags.NO_UPDATE: 'Ions are not moved',
+        IbrionFlags.MOLECULAR_DYNAMICS: 'Standard ab-initio molecular dynamics',
+        IbrionFlags.IONIC_RELAXATION_RMM_DIIS: 'Ionic relaxation using RMM-DIIS (quasi-newton algorithm)',
+        IbrionFlags.IONIC_RELAXATION_CG: 'Ionic relaxation using conjugate-gradient algorithm',
+        IbrionFlags.IONIC_RELAXATION_DAMPED_MD: 'Ionic relaxation using damped second order equation of motion',
+        IbrionFlags.HESSIAN_FD_FULL: 'Calculate Hessian using finite-differences, full trial displacements, selective dynamics supported',
+        IbrionFlags.HESSIAN_FD_SYMMETRIZED: 'Calculate Hessian using finite-differences, symmetrized trial displacements',
+        IbrionFlags.HESSIAN_PT_FULL: 'Calculate Hessian using perturbation theory, full trial displacements',
+        IbrionFlags.HESSIAN_PT_SYMMETRIZED: 'Calculate Hessian using perturbation theory, symmetrized trial displacements',
+        IbrionFlags.IMPROVED_DIMER_METHOD: 'Move ions using the Improved Dimer Method'
+    }
+
+    def __init__(self, ion_updates=IbrionFlags.NO_UPDATE):
+        if not isinstance(ion_updates, IbrionFlags):
+            raise ValueError('ion_updates parameter must be of type {}'.format(IbrionFlags))
+        self._ion_updates = ion_updates
+        self.validate()
+
+    @classproperty
+    def name(self):
+        return 'IBRION'
+
+    @staticmethod
+    def _validate_value(value):
+        allowed_values = [flag.value for flag in IbrionFlags]
+        if value not in allowed_values:
+            raise ValueError('Value {} not supported for IBRION'.format(value))
+
+    def validate(self):
+        self._validate_value(self.value)
+
+    @property
+    def value(self):
+        return self._ion_updates.value
+
+    def clean(self, incar_dict):
+        errors = []
+        warnings = []
+        if self._ion_updates != IbrionFlags.NO_UPDATE:
+            if incar_dict.get('nsw', 0) == 0:
+                warnings.append('NSW (# ionic steps) was not set or set to 0 when updating ions ({})'.format(self))
+        if self._ion_updates == IbrionFlags.MOLECULAR_DYNAMICS:
+            if 'potim' not in incar_dict:
+                errors.append('POTIM must be given when running MD ({})'.format(self))
+        if self._ion_updates in IbrionFlags.ionic_relaxation_options:  # pylint: disable=unsupported-membership-test
+            if 'potim' not in incar_dict:
+                warnings.append('POTIM was not given when running ionic relaxation ({})'.format(self))
```

Comment:
I would prefer not hardcoding VASP default values in error messages, because they can change without notice from version to version.

Reply:
We should really try to avoid adding boiler plate code to set VASP defaults. As Rico says it might change. In addition it adds a lot on the maintenance side. Also, when we have the INCAR machinery up and running in the future with a masterfile I think all this will handle these things quite nicely, so let us not hardcode these things now.

Reply:
@DropD I agree with that and admit, that my suggestion was too specific. What I wanted to achieve with my comment was letting the warning give a hint about what happens if 'POTIM' is not supplied. In particular that default values will be used. Of course on second thought that could be rather obvious. So I would be also fine with leaving it as it is.